API/src/core/ApiError.ts

```diff
@@ -0,0 +1,102 @@
+import { Response } from 'express';
+import {
+    ApiResponse,
+    AuthFailureResponse,
+    InternalErrorResponse,
+    NotFoundResponse,
+} from './ApiResponse';
+
+enum ErrorType {
+    BAD_TOKEN = 'BadTokenError',
+    TOKEN_EXPIRED = 'TokenExpiredError',
+    UNAUTHORIZED = 'AuthFailureError',
+    ACCESS_TOKEN = 'AccessTokenError',
+    INTERNAL = 'InternalError',
+    NOT_FOUND = 'NotFoundError',
+    NO_ENTRY = 'NoEntryError',
+    NO_DATA = 'NoDataError',
+    BAD_REQUEST = 'BadRequestError',
+    FORBIDDEN = 'ForbiddenError',
+}
+
+abstract class ApiError extends Error {
+    constructor(
+        public type: ErrorType,
+        private response: ApiResponse,
+        public message: string = 'error',
+    ) {
+        super(type);
+    }
+
+    public handle(res: Response): Response {
+        return this.response.send(res);
+    }
+}
+
+export class BadTokenError extends ApiError {
```

Comment:
we're not going to be adding auth to the api for a while I expect, so maybe it's better to leave the token errors out for now to not add unused code